core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala

```diff
@@ -121,16 +121,14 @@ object Main extends {
     )
   }
 
-  private def installProtocolVersion(client: skuber.api.client.KubernetesClient)(implicit ec: ExecutionContext): Unit = {
+  private def installProtocolVersion(client: skuber.api.client.KubernetesClient, podNamespace: String)(implicit ec: ExecutionContext): Unit = {
```

Comment:
This function now uses two different K8s clients, if we re-write it this way, we can do away with the risk of this problem repeating in the future.
```
private def installProtocolVersion(client: skuber.api.client.KubernetesClient)(implicit ec: ExecutionContext): Unit = {
    val protocolVersionTimeout = 20.seconds

    Await.ready(
      client.getOption[ConfigMap](Operator.ProtocolVersionConfigMapName).map {
        _.fold(client.create(Operator.ProtocolVersionConfigMap)) { configMap ⇒
          if (configMap.data.getOrElse(Operator.ProtocolVersionKey, "") != Operator.ProtocolVersion) {
            client.update(configMap.copy(data = Map(Operator.ProtocolVersionKey -> Operator.ProtocolVersion)))
          } else {
            Future.successful(configMap)
          }
        }
      },
      protocolVersionTimeout
    )
  }
```

Reply:
The function call would look like this after the change:

```
installProtocolVersion(client.usingNamespace(settings.podNamespace))
```